proposals/0002-gep.md

```diff
@@ -0,0 +1,51 @@
+---
+MEP: 0002
+Title: Unifed API
+Discussion: Link
+Implementation: Link
+---
+
+# TITLE: Unified API
+
+## Abstract
+
+This GEP discusses the unified API that handles the interaction with LLM providers.
+
+## Motivation
+
+[TBU, what problem we are trying to solve, focus on users, use cases, and all important context]
+
+The Unified API holds all the logic needed to create a request to a provider. This solution allows the clients to be easily
```

Comment:
I like to think about Unified API as a way to hide/decouple what exactly LLM provider is going to handle application/client requests. This way our end users could joggling LLM providers supported by Glide with little to no changes on their application side.

Reply:
I agree with this but we want to give them flexibility to pass specifics if they want to .

Reply:
^I see you've noted this below.

Reply:
> I agree with this but we want to give them flexibility to pass specifics if they want to

@mkrueger12 Then they change that on the gateway config end making no change to their business application. That's a shift in the mental model that LLM gateway gives. You are not doing everything directly, but you abstract away from LLM providers through the gateway. So you need to change the provider pool that serve your chat API needs you do it on the gateway side.

At the same time, if there is any benefits of doing provider-specific optimizations, we should allow it. However, it's not the same as repeating the provider pool config in the client request. I don't think we should do that.

Let know if this makes sense.